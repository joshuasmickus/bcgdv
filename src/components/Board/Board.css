.add-button,
.new-idea {
  background: none;
  border: 0;
  display: block;
  height: 40px;
  padding: 0;
  width: 40px;
}
