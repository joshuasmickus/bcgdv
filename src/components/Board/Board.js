import React, { PureComponent, Fragment } from 'react';
import { arrayOf, func, shape } from 'prop-types';

import { ideaShape } from '../../Constants/PropTypes';
import { getIdeas, newIdea } from '../../redux/ideas';

import Button from '../Button';
import Ideas from '../Ideas';

import addButton from './add.png';

export class Board extends PureComponent {
  componentDidMount() {
    const { actions, ideas } = this.props;

    ideas.length === 0 && actions.getIdeas();
  }

  render() {
    const { actions, ideas } = this.props;

    return (
      <Fragment>
        <h1>Idea Board</h1>
        <Button onClick={actions.newIdea} className="new-idea">
          <img src={addButton} alt="add-button" className="add-button" />
        </Button>
        <Ideas ideas={ideas} />
      </Fragment>
    );
  }
}

export const mapDispatchToProps = dispatch => ({
  actions: {
    getIdeas: () => dispatch(getIdeas.request()),
    newIdea: () => dispatch(newIdea.request())
  }
})

export const mapStateToProps = ({ ideas }) => ({
  ideas
});

Board.propTypes = {
  actions: shape({
    getIdeas: func.isRequired,
    newIdea: func.isRequired
  }).isRequired,
  ideas: arrayOf(shape(ideaShape))
};
