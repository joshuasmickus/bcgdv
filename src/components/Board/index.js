import { connect } from 'react-redux';

import { Board, mapDispatchToProps, mapStateToProps } from './Board';

import './Board.css';

export default connect(mapStateToProps, mapDispatchToProps)(Board);
