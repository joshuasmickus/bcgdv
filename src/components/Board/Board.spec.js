import React from 'react';
import { shallow } from 'enzyme';

import { Board } from './Board';

const getIdeasMock = jest.fn();
const newIdeaMock = jest.fn();
const actions = {
  getIdeas: getIdeasMock,
  newIdea: newIdeaMock
};

describe('Given a Board component', () => {
  describe('when it is rendered', () => {
    beforeEach(() => {
      getIdeasMock.mockReset();
    });

    it('should match the snapshot', () => {
      const wrapper = shallow(<Board actions={actions} ideas={[]} />);

      expect(wrapper.getElement()).toMatchSnapshot();
    });

    describe('and there are NO ideas', () => {
      it('should trigger getIdeas action', () => {
        shallow(<Board actions={actions} ideas={[]} />);

        expect(getIdeasMock).toHaveBeenCalledWith();
      });
    });

    describe('and there are ideas', () => {
      it('should trigger getIdeas action', () => {
        shallow(<Board actions={actions} ideas={[{}]} />);

        expect(getIdeasMock).not.toHaveBeenCalledWith();
      });
    });
  });
});
