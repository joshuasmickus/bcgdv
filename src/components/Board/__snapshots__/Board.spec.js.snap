// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a Board component when it is rendered should match the snapshot 1`] = `
<React.Fragment>
  <h1>
    Idea Board
  </h1>
  <Button
    className="new-idea"
    onClick={[MockFunction]}
  >
    <img
      alt="add-button"
      className="add-button"
      src="add.png"
    />
  </Button>
  <Ideas
    ideas={Array []}
  />
</React.Fragment>
`;
